Tools/Scripts/webkitpy/style/checkers/basexcconfig.py

```diff
@@ -0,0 +1,85 @@
+# OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+# SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+# LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+# DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
+# THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+# (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+# OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+
+from webkitpy.common.memoized import memoized
+
+
+"""Enforces rules for Base.xcconfig files."""
+
+
+class BaseXcconfigChecker(object):
+    categories = set([
+        'basexcconfig/missing-wk_default-prefix',
+        'basexcconfig/missing-inherited',
+    ])
+
+    def __init__(self, file_path, handle_style_error):
+        self._file_path = file_path
+        self._handle_style_error = handle_style_error
+
+    @memoized
+    def read_common_base_xcconfig_variables(self):
+        default_vars = {}
+        inherited_vars = {}
+
+        for line in open('Configurations/CommonBase.xcconfig'):
+            # Find lines containing assignments.
+            lhs, operator, rhs = map(str.strip, line.partition('='))
+
+            # Skip non-assignment lines, comments, and WK_ variables which are allowed to be overridden.
+            if operator != '=' or lhs.startswith('//') or lhs.startswith('WK_'):
+                continue
+
+            # Discard any setting condition.
+            name = lhs.partition('[')[0]
+
+            if '$(inherited)' in rhs:
+                inherited_vars[name] = 1
+            else:
+                default_vars[name] = 1
+
+        return default_vars, inherited_vars
+
+    def check(self, lines):
+        (default_vars, inherited_vars) = self.read_common_base_xcconfig_variables()
+
+        for line_number, line in enumerate(lines, start=1):
+            # Find lines containing assignments.
+            lhs, operator, rhs = map(str.strip, line.partition('='))
+
+            # Skip non-assignment lines, comments, and WK_ variables which are allowed to be overridden.
+            if operator != '=' or lhs.startswith('//') or lhs.startswith('WK_'):
+                continue
+
+            # Discard any setting condition.
+            name = lhs.partition('[')[0]
+
+            if default_vars.get(name):
+                self._handle_style_error(
+                    line_number, 'basexcconfig/missing-wk_default-prefix', 5,
+                    f'{name} must be declared as WK_DEFAULT_{name} (see CommonBase.xcconfig)')
+            elif inherited_vars.get(name):
+                if '$(inherited)' not in rhs and f'$(WK_COMMON_{name})' not in rhs:
+                    self._handle_style_error(
+                        line_number, 'basexcconfig/missing-inherited', 5,
+                        f'{name} must include \'$(inherited)\' or \'$(WK_COMMON_{name})\' (see CommonBase.xcconfig)')
```

Comment:
Nice! I'm glad this worked out.

Reply:
BTW, I think `GCC_OPTIMIZATION_LEVEL` is kind of unique among variables that may get moved into `CommonBase.xcconfig` in that it has a way to be overridden by Sanitizer builds.

Other settings could be moved into `CommonBase.xcconfig` but not have an override, and the variables appearing in a project's `Base.xcconfig` would just be a mistake, not a case where `WK_DEFAULT_` would need to be added.

So it may be best to move `GCC_OPTIMIZATION_LEVEL` into its own local variable as an overridable value, but I can't write a test non-overridden variables because we don't have an example of that yet.

Just something to keep in mind for the future.

Reply:
> So it may be best to move GCC_OPTIMIZATION_LEVEL into its own local variable as an overridable value, but I can't write a test non-overridden variables because we don't have an example of that yet.

I went ahead and fixed this.  New patch incoming.